src/backend/executor/process.ts

```diff
@@ -130,6 +153,41 @@ export class ExecutorProcess {
         ].join(' ');
     }
 
+    public getLogCmdLine(): string | undefined {
+        if (!workspace.workspaceFolders?.length) {
+            return undefined;
+        }
+
+        const workspaceFolder = workspace.workspaceFolders[0].uri.fsPath;
+
+        const getConfigAndReplaceVariables = (category: string, name: string): string | undefined => {
+            const configValue = workspace.getConfiguration(category).get<string>(name);
+            return configValue
+                ?.replace(/\${workspaceRoot}/g, workspaceFolder)
+                .replace(/\${workspaceFolder}/g, workspaceFolder)
+                .replace(/\${cwd}/g, process.cwd())
+                .replace(/\${env\.([^}]+)}/g, (sub: string, envName: string) => process.env[envName] ?? '');
+        };
+
+        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
+            ?? 'CodeChecker';
+        const ccFolder = getConfigAndReplaceVariables('codechecker.backend', 'outputFolder')
+            ?? path.join(workspaceFolder, '.codechecker');
+        const ccArguments = getConfigAndReplaceVariables('codechecker.executor', 'arguments') ?? '';
+        const ccThreads = workspace.getConfiguration('codechecker.executor').get<string>('threadCount');
+
+        // Use a predefined path here
+        const ccCompileCmd = path.join(ccFolder, 'compile_cmd.json');
+
+        return [
+            `${ccPath} log`,
+            `--output "${ccCompileCmd}"`,
+            `${ccThreads ? '-j ' + ccThreads : ''}`,
+            `${ccArguments}`,
+            '--build "make"'
```

Comment:
This has to be configurable because not all the projects runs simply make. For example if someone wants to compile a single make target it is not possible for now.

Above I mentioned to create a separate argument for analysis and in this case it could be the default value of it.

Reply:
It's configurable, as the user is presented a terminal where they can edit the command before running it.
The command here is only meant as a guideline, and 'make' is the one that's most common, I think.